test/metrics_test.go

```diff
@@ -182,3 +187,53 @@ func TestMetricsAuto(t *testing.T) {
 		t.Errorf("Expected value %s for %s, but got %s", "1", metricName, got)
 	}
 }
+
+// Show that when 2 blocs share the same metric listener (they have a prometheus plugin on the same listening address),
+// ALL the metrics of the second bloc in order are declared in prometheus, especially the plugins that are used ONLY in the second bloc
+func TestMetricsSeveralBlocs(t *testing.T) {
```

Comment:
At some point this needs to be moved to proper test infra that prometheus now provides

Reply:
Hum .. I am not sure what you mean.
There is a repo https://github.com/prometheus/test-infra, but that is a forked of Kubernetes/test-infra. And maybe the Team expect to adapt for Prometheus, but that is not yet done.

Reply:
iirc there is prometheus package where you can just read the metrics